src/Gateway/Braintree/SubscriptionBuilder.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace Laravel\Cashier\Gateway\Braintree;
+
+use Braintree\Customer;
+use Braintree\Subscription as BraintreeSubscription;
+use Carbon\Carbon;
+use Laravel\Cashier\Gateway\SubscriptionBuilder as BaseBuilder;
+
+class SubscriptionBuilder extends BaseBuilder
+{
+    /**
+     * The Braintree customer for this subscription.
+     *
+     * @var \Braintree\Customer
+     */
+    protected $braintreeCustomer;
+
+    /**
+     * @param  string $token
+     * @param  array $options
+     * @return static
+     */
+    public function customer($token, array $options = [])
+    {
+        $this->braintreeCustomer = $this->getBraintreeCustomer($token, $options);
+
+        return $this;
+    }
+
+    /**
+     * Create a new Braintree subscription.
+     *
+     * @param  string|null $token
+     * @param  array $subscriptionOptions
+     * @return \Laravel\Cashier\Subscription
+     * @throws \Laravel\Cashier\Gateway\Braintree\Exception
+     */
+    public function create($token = null, array $subscriptionOptions = [])
+    {
+        $this->setBraintreeCustomerFromArgs(func_get_args());
```

Comment:
Rather than passing the customer data to `create` you now chain `->customer()->create()`.